pkg/batchmapper/interface.go

```diff
@@ -0,0 +1,37 @@
+package batchmapper
+
+import (
+	"context"
+	"time"
+)
+
+// Datum contains methods to get the payload information.
+type Datum interface {
+	// Value returns the payload of the message.
+	Value() []byte
+	// EventTime returns the event time of the message.
+	EventTime() time.Time
+	// Watermark returns the watermark of the message.
+	Watermark() time.Time
+	// Headers returns the headers of the message.
+	Headers() map[string]string
+	// Id returns the unique ID set for the given message
+	Id() string
+	// Keys returns the keys associated with a given datum
+	Keys() []string
+}
+
+// BatchMapper is the interface for a Batch Map mode where the user is given a list
+// of messages, and they return the consolidated response for all of them together.
+type BatchMapper interface {
```

Comment:
Naming convention - we have `MapStreamer`.

Reply:
This makes more sense, we should rename existing `MapStreamer` to `StreamMapper`, similarly, `ReduceStreamer` -> `StreamReducer`.

Reply:
> This makes more sense, we should rename existing
why do you think this makes more sense, just curious :)

Reply:
We discussed and felt that having the operation (ie Map, Reduce, Transform) -> to be the action here.
So keeping mapper, reducer, transformer would be consistent. And the the prefix can be the special type if any.
StreamMapper, BatchMapper, StreamReducer, SessionReducer

Reply:
MapBatcher doesn't make much sense to me :)

Reply:
As discussed will create new issues to rename the previous implementations